### ISSUE TITLE
zellij 0.12 does not respond to input

### ISSUE THREAD

**Issue by cswinter** [zellij-org/zellij#538]
I just upgraded to zellij 0.12 and zellij sessions are completely unresponsive to input. None of the commands work, nor do other keys have observable effects. I've tried all of Powershell, Windows Terminal, and WSL Terminal and run zellij both under WSL and in an ssh session on an Ubuntu machine. Downgrading to 0.11 fixes the issue.

OS version:
```
Linux Nexus 5.4.0-73-generic #82-Ubuntu SMP Wed Apr 14 17:39:42 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

zellij version:
```
zellij 0.12.0
```

**Comment by imsnif** [comment]
Hey @cswinter, sorry for the experience. v0.12 moved to a multi-process model to support session detaching and my guess is that this is somehow related. Since I can't reproduce this issue, would you be willing to do some back-and-forth to help debug this? This would involve me attaching a version of zellij with some debug instructions, asking you to run it, press some keys and then attach a log file it will create.

**Comment by kunalmohan** [comment]
The problem seems to be with `termbg`. It's readme says it doesn't support Windows Terminal. But I wasn't sure about whether it works on WSL terminal, so it tried using it and it failed to get the terminal background colours. Moreover, it holds onto the stdin and so zellij can't lock onto stdin and hence doesn't respond.

**Comment by kunalmohan** [comment]
I couldn't reproduce this on my ubuntu system. But I have windows as well (I barely use it), so I tried running zellij on WSL, and was able to reproduce it there.

**Comment by imsnif** [comment]
Right... do you think there's a way we can identify windows terminal and not run termbg somehow?

**Comment by kunalmohan** [comment]
I think we can try reading the environment variable `"WT_SESSION"`. It should return `Ok()` on windows terminal only. 
That's how termbg does it
https://github.com/dalance/termbg/blob/4957dcd7d7492ac130fb9cecf10a0e82d45f394b/src/lib.rs#L95

**Comment by imsnif** [comment]
Much appreciated, @kunalmohan !

**Comment by riggs-** [comment]
This is not limited to windows. I observe the same on FreeBSD.

**Comment by imsnif** [comment]
Thanks @riggs- !
Sounds to me like we might want to ditch termbg for now while we sort this out.
@kunalmohan - do you agree? I just want to make sure that if we remove termbg completely the issue is gone. If so I'll issue a hotfix to main and ask all people involved in the thread to test it before I release a patch.

**Comment by imsnif** [comment]
Also maybe open an issue in termbg. I would guess this loop never ends even after the timeout is reached: https://github.com/dalance/termbg/blob/c451055cdba21219d0cd7d5023fd9603b554124e/src/lib.rs#L211

**Comment by kunalmohan** [comment]
@imsnif sounds good to me. I just created an issue there: https://github.com/dalance/termbg/issues/5.

**Comment by imsnif** [comment]
Right, I pushed a hotfix to main. @kunalmohan, @riggs- (and @cswinter if you're around) - could you check to make sure it fixes the issue for you?

**Comment by kunalmohan** [comment]
@imsnif works for me.

**Comment by dalance** [comment]
@imsnif  I tried to fix termbg. Could you try https://github.com/dalance/termbg/commit/fed9852c08737ab86010acc1f577be99388440e6 ?

**Comment by dalance** [comment]
@imsnif  I released termbg v0.3.0 to fix this issue.

**Comment by dalance** [comment]
I can provide such API. But I think there are some problems.

* How the user of API judges the end of `response_from_stdin`. ( termbg checks terminator character by byte )
* termbg supports not only Xterm query but also Windows API. The new API may be inconsistent with Windows API.

I'll consider the API design.

**Comment by imsnif** [comment]
Fair points. We already handle similar issues with OSC terminators elsewhere, so at least for us this is just a matter of re-using that code.

**Comment by imsnif** [comment]
@cswinter @riggs- @kebot - this has been fixed in `0.12.1` - apologies for the inconvenience. If you're still having issues, please ping me and I'll re-open and find the cause.

**Comment by riggs-** [comment]
> @cswinter @riggs- @kebot - this has been fixed in `0.12.1` - apologies for the inconvenience. If you're still having issues, please ping me and I'll re-open and find the cause.

Confirmed: This solves the problem on FreeBSD. Thank you for the quick response and fix!

**Comment by cswinter** [comment]
Can also confirm this is now working, thanks for the quick fix. So long, tmux!
